src/ComputeMetricTensorElemAlgorithm.C

```diff
@@ -0,0 +1,144 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 National Renewable Energy Laboratory.                  */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+// nalu
+#include <Algorithm.h>
+#include <ComputeMetricTensorElemAlgorithm.h>
+
+#include <FieldTypeDef.h>
+#include <Realm.h>
+#include <master_element/MasterElement.h>
+
+// stk_mesh/base/fem
+#include <stk_mesh/base/BulkData.hpp>
+#include <stk_mesh/base/Field.hpp>
+#include <stk_mesh/base/FieldParallel.hpp>
+#include <stk_mesh/base/GetBuckets.hpp>
+#include <stk_mesh/base/MetaData.hpp>
+#include <stk_mesh/base/Part.hpp>
+
+namespace sierra {
+namespace nalu {
+
+//==========================================================================
+// Class Definition
+//==========================================================================
+// ComputeMetricTensorElemAlgorithm - Metric Tensor
+//==========================================================================
+//--------------------------------------------------------------------------
+//-------- constructor -----------------------------------------------------
+//--------------------------------------------------------------------------
+ComputeMetricTensorElemAlgorithm::ComputeMetricTensorElemAlgorithm(
+    Realm &realm, stk::mesh::Part *part)
+    : Algorithm(realm, part) {
+  // save off data
+  stk::mesh::MetaData &meta_data = realm_.meta_data();
+  coordinates_ = meta_data.get_field<VectorFieldType>(
+      stk::topology::NODE_RANK, realm_.get_coordinates_name());
+  Mij_ = meta_data.get_field<GenericFieldType>(stk::topology::ELEMENT_RANK,
+                                               "metric_tensor");
+}
+
+//--------------------------------------------------------------------------
+//-------- execute ---------------------------------------------------------
+//--------------------------------------------------------------------------
+void ComputeMetricTensorElemAlgorithm::execute() {
```

Comment:
Is it necessary to do any regularization/smoothing on mij for the tensor eddy viscosity model when the mesh has poor regularity?  This would apply to gij too, I guess.

Reply:
That is unclear right now and one of the things that will need investigating when we do tests in Nalu. UT-A is not reporting that this is necessary so far.